connectors/src/connectors/google_drive/temporal/spreadsheets.ts

```diff
@@ -80,38 +79,29 @@ async function upsertTable(
     },
     truncate: true,
     parents: [tableId, ...parents],
+    useAppForHeaderDetection: true,
   });
 
   logger.info(loggerArgs, "[Spreadsheet] Table upserted.");
 }
 
 function findDataRangeAndSelectRows(allRows: string[][]): string[][] {
   // Find the first row with data to determine the range.
-  const firstNonEmptyRow = allRows.find((row) =>
+  const nonEmptyRow = allRows.filter((row) =>
     row.some((cell) => cell.trim() !== "")
   );
-  if (!firstNonEmptyRow) {
-    return []; // No data found.
-  }
-
-  // Identify the range of data: Start at the first non-empty cell and end at the nearest following empty cell or row end.
-  const startIndex = firstNonEmptyRow.findIndex((cell) => cell.trim() !== "");
```

Comment:
We have a lot of spreadsheets which fall back in this case - ( empty cells - some stuff - empty cells ) - this should not be a blocker if we use the app as the names will be found automatically. If we strip the columns at this stage, no data is passed to the worker and can't extract any name.
Even if we don't use the app, it does not break as we replace the empty cols with "unknown" - but we still keep the other columns, which is still better. 
For example : https://docs.google.com/spreadsheets/d/12xxRgNOmKBFoldntYZOld8wCw7blMcuJHxMemeXVmGg/edit?gid=0#gid=0 , only one column is currently indexed, because A2-B2 and D2 are empty - but E2/G2/H2 aren't and could still be used.